di/src/main/java/com/senlainc/warsaw/tyurin/Injector.java

```diff
@@ -0,0 +1,69 @@
+package com.senlainc.warsaw.tyurin;
+
+import com.senlainc.warsaw.tyurin.annotation.ConfigProperty;
+import com.senlainc.warsaw.tyurin.annotation.DependencyComponent;
+
+import javax.lang.model.type.PrimitiveType;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Field;
+import java.util.Properties;
+
+public class Injector {
+
+    private Properties properties;
+
+    public Injector() {
+        properties = new Properties();
+        readProperties();
+    }
+
+    private void readProperties() {
+        try (InputStream input = Injector.class.getClassLoader().getResourceAsStream("application.properties")) {
+            properties.load(input);
+        } catch (IOException e) {
+            System.out.println("Exception occurred during reading from file");
+        }
+    }
+
+    public void injectConfigProperties(Object object) throws Exception {
+        Class<?> implementationClass = object.getClass();
+        for (Field field : implementationClass.getDeclaredFields()) {
+            ConfigProperty annotation = field.getAnnotation(ConfigProperty.class);
+            if (annotation != null) {
+                Object value = customConverter(field, properties.getProperty(annotation.propertyKey()));
+                field.setAccessible(true);
+                field.set(object, value);
+            }
+        }
+    }
+
+    public void injectClasses(Object object, ApplicationContext context) throws Exception {
+        for (Field field : object.getClass().getDeclaredFields()) {
+            if (field.isAnnotationPresent(DependencyComponent.class)) {
+                field.setAccessible(true);
+                Object objectFromContext = context.getObject(field.getType());
+                field.set(object, objectFromContext);
+            }
+        }
+    }
+
+    private Object customConverter(Field field, String variable){
```

Comment:
Oh right. You can not pass class<?> to switch. This is why you used strings. My bad, it was well written before. But you can leave this like this or go back to switch with string, that's up to you